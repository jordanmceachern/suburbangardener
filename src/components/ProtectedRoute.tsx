"use client";

import React, { useEffect } from "react";
import { useRouter } from "next/navigation";
import { useAuth } from "../lib/auth-nextauth";
import LoadingSpinner from "./LoadingSpinner";

interface ProtectedRouteProps {
  children: React.ReactNode;
  redirectTo?: string;
}

export default function ProtectedRoute({
  children,
  redirectTo = "/login",
}: ProtectedRouteProps) {
  const { isAuthenticated, isLoading } = useAuth();
  const router = useRouter();

  useEffect(() => {
    if (!isLoading && !isAuthenticated) {
      router.push(redirectTo);
    }
  }, [isAuthenticated, isLoading, router, redirectTo]);

  if (isLoading) {
    return (
      <main className="min-h-screen flex items-center justify-center bg-background-light dark:bg-background-dark">
        <LoadingSpinner />
      </main>
    );
  }

  if (!isAuthenticated) {
    return null; // Will redirect
  }

  return <>{children}</>;
}
