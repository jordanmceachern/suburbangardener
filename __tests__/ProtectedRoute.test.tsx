import { render } from "@testing-library/react";
import ProtectedRoute from "@/components/ProtectedRoute";

// Mock next-auth
jest.mock("next-auth/react", () => ({
  useSession: () => ({
    data: { user: { name: "Test User" } },
    status: "authenticated",
  }),
}));

// Mock the auth hook
jest.mock("../src/lib/auth-nextauth", () => ({
  useAuth: () => ({
    isAuthenticated: true,
    isLoading: false,
  }),
}));

describe("ProtectedRoute Component", () => {
  it("should match DOM snapshot with authenticated user", () => {
    const { container } = render(
      <ProtectedRoute>
        <div>Protected content</div>
      </ProtectedRoute>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for protected content", () => {
    const { container } = render(
      <ProtectedRoute>
        <div>Protected content</div>
      </ProtectedRoute>
    );

    // Check that content is rendered
    expect(container.textContent).toContain("Protected content");
  });
});
