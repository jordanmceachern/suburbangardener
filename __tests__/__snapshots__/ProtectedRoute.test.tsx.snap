// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ProtectedRoute Component should match DOM snapshot with authenticated user 1`] = `
<div>
  Protected content
</div>
`;
